pydm/widgets/base.py

```diff
@@ -498,6 +524,115 @@ def force_redraw(self):
         """
         self.update()
 
+    @pyqtSlot(dict)
+    def setRule(self, payload):
+        """
+        Callback called when a rule has a new value for a property.
+
+        Parameters
+        ----------
+        payload : dict
+            Dictionary containing the rule name, the property to be set and the
+            new value.
+
+        Returns
+        -------
+        None
+        """
+        name = payload.get('name', '')
+        prop = payload.get('property', '')
+        value = payload.get('value', None)
+
+        if prop not in PyDMWidget.RULE_PROPERTIES:
+            logger.error('Error at Rule: %s. %s is not part of this widget properties.',
+                         name, prop)
+            return
+
+        method_name, data_type = PyDMWidget.RULE_PROPERTIES[prop]
+        method = getattr(self, method_name)
+        method(value)
+
+    @pyqtProperty(str)
+    def rules(self):
+        """
+        JSON-formatted list of dictionaries, with rules for the widget.
+
+        Returns
+        -------
+        str
+        """
+        return self._rules
+
+    @rules.setter
+    def rules(self, new_rules):
+        """
+        JSON-formatted list of dictionaries, with rules for the widget.
+
+        Parameters
+        ----------
+        new_rules : str
+
+        Returns
+        -------
+        None
+        """
+        if new_rules != self._rules:
+            self._rules = new_rules
+
+            # Let's clean up the current actions
+            # and terminate all the action threads
+            # that are there...
+            if len(self._rules_objs) != 0:
+                for ao in self._rules_objs:
```

Comment:
Will this deconstruct the entire rules list then rebuild it if we append a single rule?

Reply:
Yes! I could transform the `new_rules` into a list and compare again... check what we are missing and destroy and check what changed and create a new one and what was added but for simplicity I just destroy all and recreate...
What do you think?